Resources/templates/responsive/project/participate.php

```diff
@@ -1,4 +1,8 @@
-<?php $this->layout('project/layout') ?>
+<?php
+
+use Goteo\Payment\Payment;
+
+ $this->layout('project/layout') ?>
```

Comment:
I think i would prefer to have a function inside `Invest.php` that can tell us if it's a subscription or not. This way we could not import any class and just use `$invest->isSubscriptionMethod()`

```php
public function isSubscriptionMethod() 
{
  return self::METHOD_STRIPE_SUBSCRIPTION == $this->method
}
```

How do you see it?

Reply:
`Invest`s are a byproduct of a `Payment`, which can be of many shapes and types, the logic for checking if a payment method is subscription-based or not is, by design, more complex than just checking for Stripe as we might have more subscription-based payment options in the future.

I chose to leave that logic outside of the `Invest` object and into the `Payment` class to help with a separation of concerns and not enlarge an entity that is already thousand of code lines long. Hope you can agree this solution is "the less bad one".

Reply:
I don't know... `Invest` already has functions to ask about the payments methods involved. I know the solution I posted is not scalable.

Since we can ask the Invest for their payment method, we could then ask it if it's a subscription method.

```php
<?= $invest->getMethod()->isSubscriptionMethod() ? $this->text('project-invest-from-subscription') : $this->text('project-invest') ?>
```

```php AbstractPaymentMethod
   public function isSubscriptionMethod(): bool
   {
       return Payment::isSubscriptionMethod($this->getIdNonStatic());
   }
```

I just don't really like seeing imports inside the views tbh.

Still, if you don't see it, we can stick with the development you did.

Reply:
I like this one too. Will implement.